test/file-operations.js

```diff
@@ -333,6 +362,206 @@ describe('getTimesDiff', function() {
+
+    done();
+  });
+
+  it('returns undefined if vinyl uid & gid are both equal to counterparts', function(done) {
+    var fsStat = {
+      uid: 1000,
+      gid: 1000,
+    };
+    var vfsStat = {
+      uid: 1000,
+      gid: 1000,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(undefined);
+
+    done();
+  });
+
+  it('returns a diff object if uid or gid do not match', function(done) {
+    var fsStat = {
+      uid: 1000,
+      gid: 1000,
+    };
+    var vfsStat = {
+      uid: 1001,
+      gid: 1000,
+    };
+    var expected = {
+      uid: 1001,
+      gid: 1000,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(expected);
+
+    vfsStat = {
+      uid: 1000,
+      gid: 1001,
+    };
+    expected = {
+      uid: 1000,
+      gid: 1001,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(expected);
+
+    done();
+  });
+
+  it('returns the fs uid if the vinyl uid is invalid', function(done) {
+    var fsStat = {
+      uid: 1000,
+      gid: 1000,
+    };
+    var vfsStat = {
+      uid: undefined,
+      gid: 1001,
+    };
+    var expected = {
+      uid: 1000,
+      gid: 1001,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(expected);
+
+    var vfsStat = {
+      uid: -1,
+      gid: 1001,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(expected);
+
+    done();
+  });
+
+  it('returns the fs gid if the vinyl gid is invalid', function(done) {
+    var fsStat = {
+      uid: 1000,
+      gid: 1000,
+    };
+    var vfsStat = {
+      uid: 1001,
+      gid: undefined,
+    };
+    var expected = {
+      uid: 1001,
+      gid: 1000,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(expected);
+
+    var vfsStat = {
+      uid: 1001,
+      gid: -1,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(expected);
+
+    done();
+  });
+
+  it('makes uid diff undefined if fs and vinyl uid are invalid', function(done) {
+    var fsStat = {
+      uid: undefined,
+      gid: 1000,
+    };
+    var vfsStat = {
+      uid: undefined,
+      gid: 1001,
+    };
+    var expected = {
+      uid: undefined,
+      gid: 1001,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(expected);
+
+    var fsStat = {
+      uid: -1,
+      gid: 1000,
+    };
+    var vfsStat = {
+      uid: -1,
+      gid: 1001,
+    };
+
+    var result = getOwnerDiff(fsStat, vfsStat);
+
+    expect(result).toEqual(expected);
+
+    done();
+  });
+
+  it('makes gid diff undefined if fs and vinyl gid are invalid', function(done) {
+    var fsStat = {
+      uid: 1000,
+      gid: undefined,
+    };
+    var vfsStat = {
+      uid: 1001,
+      gid: undefined,
+    };
+    var expected = {
+      uid: 1001,
+      gid: undefined,
```

Comment:
fs.fchown throws if uid or gid is undefined.  Do we need to handle this by returning `undefined` if we don't have a complete diff?